src/librustc_resolve/diagnostics.rs

```diff
@@ -93,6 +93,36 @@ mod bar {
 ```
 "##,
 
+E0254: r##"
+This error indicates that a `use` declaration added a symbol to the current
+namespace that conflicts with a symbol added by an `extern crate` declaration.
+
+An example of this error:
+
+There's an external crate and it, also, can have no exports as foo.rs:
+
+```
+#![crate_type="lib"]
+```
+
+On the client code, bar.rs:
+
+```
+extern crate foo;
+
+use inner_mod::foo; // error, both of these `use` declarations add a symbol
+use another_mod::foo; // that conflicts with the external crate's symbol.
```

Comment:
I am not sure I would personally have put *both* of the examples errors into a single example file, since if you show it this way, then removing the `extern crate foo;` above actually fails to yield code that will compile (since these two imports are themselves in conflict).

The only reason that I included both examples in my write up to you is because you (I think) had included two distinct examples in your original draft.

I would strongly consider either putting in both as separate examples, or just removing the variant that uses `mod inner_mod { pub mod foo { } } ` and instead just use the one that exports a type.